modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala

```diff
@@ -64,71 +59,54 @@ object Execution {
   /**
     * Make an `Execution` `Current` only if all the `Action`s in the execution
     * are pending.
-    *
     */
   def currentify(as: Actions): Option[Execution] =
-    as.nonEmpty.option(Execution(as.map(_.left)))
-
-  def errored(ar: Action \/ Result): Boolean =
-    ar match {
-      case (-\/(_)) => false
-      case (\/-(r)) => r match {
-        case Result.Error(_, _) => true
-        case _                  => false
-      }
-    }
-
-  def finished(ar: Action \/ Result): Boolean =
-    ar match {
-      case (-\/(_)) => false
-      case (\/-(r)) => r match {
-        case Result.Partial(_, _) => false
-        case _                    => true
-      }
-    }
+    (as.nonEmpty && as.forall((_.state === Action.Idle))).option(Execution(as))
```

Comment:
Will `forall` be false if empty? If so the first check is not needed

Reply:
`> List[Int]().forall(_>1)
res3: Boolean = true`
The reason is that `forall` is the complement of `exists`. It is the same for `Option`. `None.forall(f)` is always `true`.